cli/command/command.go

```diff
@@ -18,11 +18,72 @@ package command
 
 import (
 	"fmt"
+	"strings"
 
+	"github.com/Sirupsen/logrus"
 	"github.com/skippbox/kompose/cli/app"
 	"github.com/urfave/cli"
 )
 
+// Hook for erroring and exit out on warning
+type errorOnWarningHook struct{}
+
+func (errorOnWarningHook) Levels() []logrus.Level {
+	return []logrus.Level{logrus.WarnLevel}
+}
+
+func (errorOnWarningHook) Fire(entry *logrus.Entry) error {
+	logrus.Fatalln(entry.Message)
+	return nil
+}
+
+// BeforeApp is an action that is executed before any cli command.
+func BeforeApp(c *cli.Context) error {
+
+	if c.GlobalBool("verbose") {
+		logrus.SetLevel(logrus.DebugLevel)
+	} else if c.GlobalBool("suppress-warnings") {
+		logrus.SetLevel(logrus.ErrorLevel)
+	} else if c.GlobalBool("error-on-warning") {
+		hook := errorOnWarningHook{}
+		logrus.AddHook(hook)
+	}
+
+	provider := strings.ToLower(c.GlobalString("provider"))
+	switch provider {
+	case "kubernetes":
+		k8sControllers := []cli.Flag{
```

Comment:
can we create whole new `cli.Command` here?  So we can set usage help for every provider independently?
Or set `commands.Usage` here

If I run `kompose --provider openshift convert -h ` It is still displaying usage mentioning Kubernetes

```
NAME:
   kompose convert - Convert Docker Compose file (e.g. docker-compose.yml) to Kubernetes objects
....
```

This is also relying on `ConvertCommand` always being  first in that slice :-( This might lead to errors in the future. At least there should be comment in `app.Commands`